src/pybind/mgr/devicehealth/module.py

```diff
@@ -399,8 +399,7 @@ def show_device_metrics(self, devid, sample):
             return -errno.ENOENT, '', 'device ' + devid + ' not found'
         # fetch metrics
         res = {}
-        ioctx = self.open_connection(create_if_missing=False)
-        if ioctx:
+        with self.open_connection(create_if_missing=False) as ioctx:
```

Comment:
@hsiang41 sorry, i just realized that if `create_if_missing` is False, and pool is not found, `open_connection()` returns None. so how about
```py
ioctx = self.open_connection(create_if_missing=False)
if ioctx is None:
    return 0, json.dumps(res, indent=4), ''
with ioctx:
    # ...
```

Reply:
Great! thanks.